src/EnergyPlus/DuctLoss.cc

```diff
@@ -0,0 +1,1060 @@
+            state.afn->AirflowNetworkNodeSimu(NodeNum2).WZ = Wout;
+        } break;
+        case DuctLossSubType::ReturnBranch: {
+            NodeNum = state.afn->DisSysNodeData(NodeNum1).EPlusNodeNum;
+            MassFlowRate = state.dataLoopNodes->Node(NodeNum).MassFlowRate;
+            Tin = state.dataLoopNodes->Node(NodeNum).Temp;
+            state.afn->AirflowNetworkNodeSimu(NodeNum1).TZ = Tin;
+            Win = state.dataLoopNodes->Node(NodeNum).HumRat;
+            state.afn->AirflowNetworkNodeSimu(NodeNum1).WZ = Win;
+            CpAir = Psychrometrics::PsyCpAirFnW(state.dataLoopNodes->Node(NodeNum).HumRat);
+            Tout = Tamb + (Tin - Tamb) * General::epexp(-UThermal * DuctSurfArea, (MassFlowRate * CpAir));
+            Wout = Wamb + (Win - Wamb) * General::epexp(-UMoisture * DuctSurfArea, MassFlowRate);
+            this->Qsen = -MassFlowRate * CpAir * (Tamb - Tin) * (1.0 - General::epexp(-UThermal * DuctSurfArea, (MassFlowRate * CpAir)));
+            this->Qlat = -MassFlowRate * (Wamb - Win) * (1.0 - General::epexp(-UMoisture * DuctSurfArea, MassFlowRate));
+            state.afn->AirflowNetworkNodeSimu(NodeNum2).TZ = Tout;
+            state.afn->AirflowNetworkNodeSimu(NodeNum2).WZ = Wout;
+            if (!state.dataDuctLoss->SubTypeSimuFlag(int(DuctLossSubType::ReturnTrunk) + 1)) {
+                state.dataLoopNodes->Node(state.dataMixerComponent->MixerCond(1).OutletNode).Temp = Tout;
+                state.dataLoopNodes->Node(state.dataMixerComponent->MixerCond(1).OutletNode).HumRat = Wout;
+            }
+        } break;
+
+        default:
+            break;
+        }
+    }
+
+    void DuctLossComp::CalcLeakage(EnergyPlusData &state)
+    {
+        Real64 MassFlowRate;
+        Real64 Tin;
+        Real64 Tout;
+        Real64 Tzone;
+        Real64 Win;
+        Real64 Wout;
+        Real64 Wzone;
+        Real64 CpAir;
+        int NodeNum1;
+        int NodeNum2;
+        int NodeNum;
+        int ZoneNodeNum;
+        NodeNum1 = state.afn->AirflowNetworkLinkageData(this->LinkageNum).NodeNums[0];
+        NodeNum2 = state.afn->AirflowNetworkLinkageData(this->LinkageNum).NodeNums[1];
+
+        int TypeNum = state.afn->AirflowNetworkCompData(state.afn->AirflowNetworkLinkageData(this->LinkageNum).CompNum).TypeNum;
+        Real64 LeakRatio = state.afn->DisSysCompELRData(TypeNum).ELR;
+
+        this->Qsen = 0.0;
+        this->Qlat = 0.0;
+        this->QsenSL = 0.0;
+        this->QlatSL = 0.0;
+
+        switch (this->LossSubType) {
+        case DuctLossSubType::SupLeakTrunk: {
+            // No need to calculate equivalent T and W. Keep the same value
+            Tin = state.afn->AirflowNetworkNodeSimu(NodeNum1).TZ;
+            Win = state.afn->AirflowNetworkNodeSimu(NodeNum1).WZ;
+            CpAir = Psychrometrics::PsyCpAirFnW(Win);
+            // Zone only
+            this->Qsen = state.dataLoopNodes->Node(state.dataDuctLoss->AirLoopInNodeNum).MassFlowRate * CpAir * LeakRatio *
```

Comment:
Use the local `MassFlowRate` to avoid repeating this?

Reply:
Done